File: src/main/java/io/enthusia/express/util/ItemCodec.java
package io.enthusia.express.util;

import org.bukkit.inventory.ItemStack;

public final class ItemCodec {
  private ItemCodec() {}

  public static byte[] encode(ItemStack stack) {
    return stack.serializeAsBytes();
  }

  public static ItemStack decode(byte[] data) {
    return ItemStack.deserializeBytes(data);
  }
}

File: src/main/kotlin/io/enthusia/express/infrastructure/util/ItemCodec.kt
package io.enthusia.express.infrastructure.util

import org.bukkit.inventory.ItemStack

object ItemCodec {
    /** Serialize an item snapshot with Paper item serialization for persistence. */
    @JvmStatic
    fun encode(stack: ItemStack): ByteArray = stack.serializeAsBytes()

    /** Restore a stored item using Paper serialization, propagating invalid-payload failures. */
    @JvmStatic
    fun decode(data: ByteArray): ItemStack = ItemStack.deserializeBytes(data)
}
